src/DuskServiceProvider.php

```diff
@@ -14,286 +14,288 @@ class DuskServiceProvider extends ServiceProvider
-        });
-
-        Browser::macro('setBrowserLocationToNewYork', function () {
-            $this->setLocationToLatLon(40.730610, -73.935242, 'America/New_York');
-        });
-
-        Browser::macro('setLocationToLatLon', function (float $lat, float $lon, string $timezone) {
-            $devTools = new ChromeDevToolsDriver($this->driver);
-
-            $location = [
-                $lat,
-                $lon,
-                1,
-            ];
-
-            $devTools->execute('Emulation.setTimezoneOverride', ['timezoneId' => $timezone]);
-            $devTools->execute('Emulation.setGeolocationOverride', $location);
-            $this->pause(100);
-        });
-
-        Browser::macro('inRow', function (string $stringInTable, \Closure $closure) {
-            $resolver = new ElementResolver(
-                $this->driver->findElement(WebDriverBy::xpath('//table//tr[td/span//text()[contains(., "' . $stringInTable . '")]]')),
-            );
-
-            $browser = new Browser(
-                $this->driver,
-                $resolver
-            );
-
-            $closure($browser);
-        });
-
-        Browser::macro('visitTwill', function () {
-            if (
-                ! Str::contains($this->driver->getCurrentURL(), '/twill') ||
-                Str::contains($this->driver->getCurrentURL(), '_dusk')
-            ) {
-                $this->visit('/twill');
-            }
-        });
-
-        Browser::macro('visitModuleEntryWithTitle', function (string $menuName, string $title) {
-            $this->visitTwill();
-
-            $this->clickLink($menuName);
-
-            $this->waitForText($title);
-            $this->clickLink($title);
-
-            $this->assertSee($title);
-        });
-
-        Browser::macro('createModuleEntryWithTitle', function (string $menuName, string $title) {
-            $this->visitTwill();
-
-            $this->waitForText($menuName);
-            $this->clickLink($menuName);
-
-            $this->createWithTitle($title);
-        });
-
-        Browser::macro('createWithTitle', function (string $title) {
-            $this->press('Add new');
-
-            $this->waitFor('.modal__header');
-
-            if ($this->element('.input-wrapper-title\[en\]')) {
-                $this->type('title[en]', $title);
-            } else {
-                $this->type('title', $title);
-            }
-
-            $this->press('Create');
-
-            $this->waitForReload();
-        });
-
-        Browser::macro('assertVselectHasOptions', function (string $wrapperClass, array $optionLabels) {
-            $this->with($wrapperClass, function (Browser $element) use ($optionLabels) {
-                $element->click('.vs__search');
-                $element->waitFor('.vs__dropdown-menu');
-
-                foreach ($optionLabels as $optionLabel) {
-                    $element->assertSeeIn('.vs__dropdown-menu', $optionLabel);
+        if (class_exists(Laravel\Dusk\Browser::class)) {
```

Comment:
Please return early by checking the inexistence instead, to avoid wrapping all that code under  a conditional. This will also make this PR diff a lot easier to review. Thanks for your contribution!

Reply:
As someone once showed me, you can use the `hide whitespace` feature to see a clean diff
![image](https://github.com/area17/twill/assets/6115458/12cfa6a2-00d2-4f4e-9706-498813dcdbd0)

Reply:
For sure! I use this too and it's great. In this case I was more concerned by the unnecessary indentation of all that code. Thanks for addressing it.